src/HostProgram.elm

```diff
@@ -125,19 +169,29 @@ src value =
     attribute "src" value
 
 
+onClientMessage : Attribute Msg
+onClientMessage =
+    on "clientMessage" (Decode.field "detail" ClientMessage.decoder |> Decode.map ClientMsg)
```

Comment:
When there's a `clientMessage` event, pull the data from the `detail` field on the event, decode it to a `ClientMessage` and tag it as `ClientMsg` to convert it to this program's `Msg` type, for handling in `update`